### ISSUE TITLE
`WritablePresignedPostRequest` improperly requiring readonly property

### ISSUE THREAD

**Issue by reefdog** [PhilanthropyDataCommons/service#1012]
When trying to use an argument of `WritablePresignedPostRequest` type — which should only require `fileType` and `fileSize` — TypeScript gets bigmad that we aren't including `presignedPost` as well.

This API-access hook should work:
```ts
const usePresignedPostCallback = () => {
	const api = usePdcCallbackApi<PresignedPostRequest>('/presignedPostRequests');
	return (params: WritablePresignedPostRequest) =>
		api({
			method: 'post',
			headers: {
				Accept: 'application/json',
				'Content-Type': 'application/json',
			},
			body: JSON.stringify(params),
		});
};
```

And would be called thusly:
```tsx
const createPresignedPost = usePresignedPostCallback();
const { presignedPost } = await createPresignedPost({
	fileType: file.type || 'application/octet-stream',
	fileSize: file.size,
});
```

However, that implementation throws this error:

> Argument of type '{ fileType: string; fileSize: number; }' is not assignable to parameter of type '{ fileType: string; fileSize: number; presignedPost: { fields: { key: string; }; url: string; }; }'.
>  Property 'presignedPost' is missing in type '{ fileType: string; fileSize: number; }' but required in type '{ fileType: string; fileSize: number; presignedPost: { fields: { key: string; }; url: string; }; }'.

Our temporary fix on the front end was to instead define and use this local type in the callback:

```ts
type FixedWritablePresignedPostRequest = Omit<
	WritablePresignedPostRequest,
	'presignedPost'
>;
```

**Comment by slifty** [comment]
Grumble grumble.

So this is properly marked as readonly: https://github.com/PhilanthropyDataCommons/service/blob/5795b15786e3c3d098af6522b8d61d5e6de3b386/src/openapi.json#L250

However it looks like swagger codegen is not honoring that fact.  I'm going to take a moment trying to figure out the state of reality with swagger codegen right now to make sure we're on the latest version.

**Comment by slifty** [comment]
Some related resources:

- https://github.com/swagger-api/swagger-codegen/issues/6537 <-- it looks like this bug was resolved, so I'm not sure it's directly related
- https://stackoverflow.com/questions/51402156/how-to-declare-a-ref-property-as-readonly-in-openapi-swagger

Some of this might also inform fixes to #1004
